src/fields/field_fe.cc

```diff
@@ -698,6 +698,14 @@ inline unsigned int FieldFE<spacedim, Value>::data_size() const {
 
 
 
+template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::local_to_ghost_data_scatter() {
```

Comment:
Replace by two inlined methods one for begin, one for end. This can be used to overlap communication with calculation.